src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs

```diff
@@ -213,43 +198,6 @@ protected override bool ReleaseHandle()
         }
     }
 
-#if DEBUG
-    internal abstract class SafeDeleteContext : DebugSafeHandle
-    {
-#else
-    internal abstract class SafeDeleteContext : SafeHandle
-    {
-#endif
-        private SafeFreeCredentials _credential;
-
-        protected SafeDeleteContext(SafeFreeCredentials credential)
-            : base(IntPtr.Zero, true)
-        {
-            Debug.Assert((null != credential), "Invalid credential passed to SafeDeleteContext");
-
-            // When a credential handle is first associated with the context we keep credential
-            // ref count bumped up to ensure ordered finalization. The credential properties
-            // are used in the SSL/NEGO data structures and should survive the lifetime of
-            // the SSL/NEGO context
-            bool ignore = false;
-            _credential = credential;
-            _credential.DangerousAddRef(ref ignore);
-        }
-
-        public override bool IsInvalid
-        {
-            get { return (null == _credential); }
-        }
-
-        protected override bool ReleaseHandle()
-        {
-            Debug.Assert((null != _credential), "Null credential in SafeDeleteContext");
-            _credential.DangerousRelease();
-            _credential = null;
-            return true;
-        }
-    }
-
     internal sealed class SafeDeleteSslContext : SafeDeleteContext
```

Comment:
Everything left in this file (for example, this class) should be also moved to new files so we can delete the whole libssl directory